src/components/smart_objects/src/object_schema_item.cc

```diff
@@ -47,9 +49,67 @@ namespace NsSmartObjects {
 CObjectSchemaItem::SMember::SMember()
     : mSchemaItem(CAlwaysFalseSchemaItem::create()), mIsMandatory(true) {}
 
-CObjectSchemaItem::SMember::SMember(const ISchemaItemPtr SchemaItem,
-                                    const bool IsMandatory)
-    : mSchemaItem(SchemaItem), mIsMandatory(IsMandatory) {}
+CObjectSchemaItem::SMember::SMember(
+    const ISchemaItemPtr SchemaItem,
+    const bool IsMandatory,
+    const std::string& Since,
+    const std::string& Until,
+    const bool IsDeprecated,
+    const bool IsRemoved,
+    const std::vector<CObjectSchemaItem::SMember>& history_vector)
+    : mSchemaItem(SchemaItem), mIsMandatory(IsMandatory) {
+  if (Since.size() > 0) {
+    utils::SemanticVersion since_struct;
+    std::vector<std::string> since_fields;
+    boost::split(since_fields, Since, boost::is_any_of("."));
+    if (since_fields.size() == 3) {
+      since_struct.major_version = atoi(since_fields[0].c_str());
+      since_struct.minor_version = atoi(since_fields[1].c_str());
+      since_struct.patch_version = atoi(since_fields[2].c_str());
+      mSince = since_struct;
+    }
+  }
+  if (Until.size() > 0) {
+    utils::SemanticVersion until_struct;
+    std::vector<std::string> until_fields;
+    boost::split(until_fields, Until, boost::is_any_of("."));
+    if (until_fields.size() == 3) {
+      until_struct.major_version = atoi(until_fields[0].c_str());
+      until_struct.minor_version = atoi(until_fields[1].c_str());
+      until_struct.patch_version = atoi(until_fields[2].c_str());
+      mUntil = until_struct;
+    }
+  }
+  mIsDeprecated = IsDeprecated;
+  mIsRemoved = IsRemoved;
+  mHistoryVector = history_vector;
+}
+
+bool CObjectSchemaItem::SMember::CheckHistoryFieldVersion(
+    const utils::SemanticVersion& MessageVersion) const {
+  if (MessageVersion.isValid()) {
+    if (mSince.is_initialized()) {
+      if (MessageVersion < mSince.get()) {
+        return false;  // Msg version predates `since` field
+      } else {
+        if (mUntil.is_initialized() && (MessageVersion >= mUntil.get())) {
+          return false;  // Msg version newer than `until` field
```

Comment:
@JackLivio nested ifs are too deep. I would refactor this method to multiple returns